### PR TITLE
Add tests to Mappers

### LINKED ISSUE #128: Create unit test to Mappers 1-5
🧩 Create Unit Tests for Mappers in greencity.mapping (GreenCityMVP)
Package:
service/src/main/java/greencity/mapping

Description
Write unit tests for all mapper classes in the greencity.mapping package.
These classes convert between entity objects, DTOs, and VO (value objects), and are essential for ensuring proper separation between data models and API representations.

Classes to Cover

CategoryDtoMapper
CategoryDtoResponseMapper
CustomHabitMapper
CustomShoppingListMapper
CustomShoppingListResponseDtoMapper

For each mapper:

Convert from entity to DTO

Convert from DTO to entity (if applicable)

Handle null or empty input

Validate that all fields are properly mapped

Assert object equality using assertEquals or deep comparison libraries
Acceptance Criteria

All tests pass successfully

Tests follow Arrange-Act-Assert structure

Mappers that use other dependencies (e.g., ModelMapper) are mocked or configured

Corner cases like null or partially filled objects are tested

100% line and branch test coverage

Code is clean and adheres to project standards
Tips:

Group similar mappers into test classes by domain (EcoNews, Habit, ShoppingList, etc.)
Use test data builders or static mock DTO/entity objects to avoid repetition
Verify the exact values of mapped fields, not just non-nullity

### PR DESCRIPTION
**Pull request by Kostya-Pav**
## Summary by CodeRabbit

* **Tests**
  * Added tests for category mapping, including validation of correct and incorrect data handling.
  * Introduced tests for converting category entities to response DTOs, covering both populated and empty cases.
  * Implemented tests for custom habit mapping, ensuring correct field transfer and handling of null values.
  * Added tests for custom shopping list item mapping, including single and batch conversions, and null field scenarios.
  * Provided tests for converting shopping list items to response DTOs, verifying both individual and list conversions.

### CODE REVIEW

**Review by Linney34 — APPROVED** [review]

**Review by natri0 — APPROVED** [review]

**Review by AndriiSynytsia — CHANGES_REQUESTED** [review]
Hey. Nice code. Test coverage 100%. There is only one thing that disturbs me. Would be great to have explanation for each test as a DisplayName annotation, it will be more informative for next contributors :)

File: service/src/test/java/greencity/mapping/CategoryDtoMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import jakarta.validation.ValidatorFactory;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.Set;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryDtoMapperTest {
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CategoryDtoMapperTest.java]: Hi, using a public modifier in the test doesn't make much sense. It's better to remove it.

File: service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryDtoResponseMapperTest {
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java]: The same problem

File: service/src/test/java/greencity/mapping/CustomHabitMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import greencity.dto.habit.AddCustomHabitDtoRequest;
+import greencity.entity.Habit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomHabitMapperTest {
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CustomHabitMapperTest.java]: The same problem

File: service/src/test/java/greencity/mapping/CustomShoppingListMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import greencity.entity.CustomShoppingListItem;
+import greencity.enums.ShoppingListItemStatus;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomShoppingListMapperTest {
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CustomShoppingListMapperTest.java]: The same problem

File: service/src/test/java/greencity/mapping/CustomShoppingListResponseDtoMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import greencity.entity.CustomShoppingListItem;
+import greencity.enums.ShoppingListItemStatus;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomShoppingListResponseDtoMapperTest {
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CustomShoppingListResponseDtoMapperTest.java]: The same problem

**Review by MykytaSirobaba — COMMENTED** [review]
In JUnit 5, test methods don’t need to be public — package-private is enough. Using public adds no value here; it's a minor issue.

File: service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package greencity.mapping;
+
+import greencity.dto.category.CategoryDto;
```
Comment (MykytaSirobaba) [inline service/src/test/java/greencity/mapping/CategoryDtoResponseMapperTest.java]: Unused import

**Review by MykytaSirobaba — CHANGES_REQUESTED** [review]

**Review by MykytaSirobaba — APPROVED** [review]
